### ISSUE TITLE
Nextcloud: fail open files in webui

### ISSUE THREAD

**Issue by ndruba** [coreruleset/coreruleset#1948]
### Description
 ````
--3974fd1a-A--
[09/Dec/2020:01:00:34 +0300] X8-3gC8t@p8dA1rIDm3ABwAAlws 192.168.0.11 52567 192.168.0.27 443
--3974fd1a-B--
PROPFIND /remote.php/dav/files/root/ HTTP/2.0
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0
Accept: */*
Accept-Language: ru-RU,ru;q=0.8,en-US;q=0.5,en;q=0.3
Accept-Encoding: gzip, deflate, br
X-Requested-With: XMLHttpRequest
Requesttoken: Ywq6rZNti+Gbyi8KhOJ5jrJL2zU8K5YYPKc2Si7jh/0=:Vj7+/Pc3uIv/nFU746BLvYAm6FcFZ/JMX8pnImua8oo=
Depth: 1
Content-Type: application/xml; charset=utf-8
Content-Length: 621
Origin: https://cloud.example.com
Dnt: 1
Cookie: __Host-nc_sameSiteCookielax=true; __Host-nc_sameSiteCookiestrict=true; nc_username=root; nc_token=%2BXZxBY9JfwBMmbmJr8LOA9PyjsCMrI5w; nc_session_id=kiiekiom9u7jb8tdt5vfnhq4nv; oc_sessionPassphrase=jXTF8AHxCsrOuQ9QkW280JAzm8UA7%2BeOJ6vrvzZV20SHXqAq88xgF0XjgB6RkbsD%2B3HdnWJqD8I4femKvxUyt7DldgkpzLQF6k8nnQc9hzK6QAlQUitVFjYiINF5Zq1f; i18next=ru-RU; och0h0q18vvv=kiiekiom9u7jb8tdt5vfnhq4nv
Te: trailers
Host: cloud.example.com

--3974fd1a-C--
<?xml version="1.0"?>
<d:propfind  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns" xmlns:ocs="http://open-collaboration-services.org/ns">
  <d:prop>
    <d:getlastmodified />
    <d:getetag />
    <d:getcontenttype />
    <d:resourcetype />
    <oc:fileid />
    <oc:permissions />
    <oc:size />
    <d:getcontentlength />
    <nc:has-preview />
    <nc:mount-type />
    <nc:is-encrypted />
    <ocs:share-permissions />
    <oc:tags />
    <oc:favorite />
    <oc:comments-unread />
    <oc:owner-id />
    <oc:owner-display-name />
    <oc:share-types />
  </d:prop>
</d:propfind>
--3974fd1a-F--
HTTP/1.1 207 Multi-Status
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate
Pragma: no-cache
Content-Security-Policy: default-src 'none';
Vary: Brief,Prefer
DAV: 1, 3, extended-mkcol, access-control, calendarserver-principal-property-search, nc-calendar-search, nc-enable-birthday-calendar
Content-Encoding: gzip
Referrer-Policy: no-referrer
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-Robots-Tag: none
X-XSS-Protection: 1; mode=block
Connection: close
Content-Type: application/xml; charset=utf-8

--3974fd1a-E--

--3974fd1a-H--
Message: Rule 802d49b60 [id "932100"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "122"] - Execution error - PCRE limits exceeded (-8): (null).
Message: Rule 802d537e0 [id "932105"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "158"] - Execution error - PCRE limits exceeded (-8): (null).
Message: Rule 802d5d6c0 [id "932110"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "255"] - Execution error - PCRE limits exceeded (-8): (null).
Message: Rule 802d65738 [id "932115"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "294"] - Execution error - PCRE limits exceeded (-8): (null).
Message: Rule 802e5abc0 [id "932150"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "463"] - Execution error - PCRE limits exceeded (-8): (null).
Message: Rule 802782378 [id "942360"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf"][line "444"] - Execution error - PCRE limits exceeded (-8): (null).
Apache-Error: [file "ssl_engine_kernel.c"] [line 376] [level 7] AH02034: Subsequent (No.8) HTTPS request received for child 38667 (server cloud.example.com:443)
Apache-Error: [file "mod_authz_core.c"] [line 815] [level 7] AH01626: authorization result of Require all granted: granted
Apache-Error: [file "mod_authz_core.c"] [line 815] [level 7] AH01626: authorization result of <RequireAny>: granted
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 192.168.0.11] ModSecurity: Rule 802d49b60 [id "932100"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "122"] - Execution error - PCRE limits exceeded (-8): (null). [hostname "cloud.example.com"] [uri "/remote.php/dav/files/root/"] [unique_id "X8-3gC8t@p8dA1rIDm3ABwAAlws"]
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 192.168.0.11] ModSecurity: Rule 802d537e0 [id "932105"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "158"] - Execution error - PCRE limits exceeded (-8): (null). [hostname "cloud.example.com"] [uri "/remote.php/dav/files/root/"] [unique_id "X8-3gC8t@p8dA1rIDm3ABwAAlws"]
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 192.168.0.11] ModSecurity: Rule 802d5d6c0 [id "932110"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "255"] - Execution error - PCRE limits exceeded (-8): (null). [hostname "cloud.example.com"] [uri "/remote.php/dav/files/root/"] [unique_id "X8-3gC8t@p8dA1rIDm3ABwAAlws"]
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 192.168.0.11] ModSecurity: Rule 802d65738 [id "932115"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "294"] - Execution error - PCRE limits exceeded (-8): (null). [hostname "cloud.example.com"] [uri "/remote.php/dav/files/root/"] [unique_id "X8-3gC8t@p8dA1rIDm3ABwAAlws"]
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 192.168.0.11] ModSecurity: Rule 802e5abc0 [id "932150"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf"][line "463"] - Execution error - PCRE limits exceeded (-8): (null). [hostname "cloud.example.com"] [uri "/remote.php/dav/files/root/"] [unique_id "X8-3gC8t@p8dA1rIDm3ABwAAlws"]
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 192.168.0.11] ModSecurity: Rule 802782378 [id "942360"][file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf"][line "444"] - Execution error - PCRE limits exceeded (-8): (null). [hostname "cloud.example.com"] [uri "/remote.php/dav/files/root/"] [unique_id "X8-3gC8t@p8dA1rIDm3ABwAAlws"]
Apache-Error: [file "mod_proxy.c"] [line 1251] [level 7] AH01143: Running scheme unix handler (attempt 0)
Apache-Error: [file "mod_proxy_fcgi.c"] [line 1030] [level 7] AH01076: url: fcgi://localhost//usr/local/www/nextcloud/remote.php proxyname: (null) proxyport: 0
Apache-Error: [file "mod_proxy_fcgi.c"] [line 1039] [level 7] AH01078: serving URL fcgi://localhost//usr/local/www/nextcloud/remote.php
Apache-Error: [file "proxy_util.c"] [line 2392] [level 7] AH00944: connecting fcgi://localhost//usr/local/www/nextcloud/remote.php to localhost:8000
Apache-Error: [file "proxy_util.c"] [line 2428] [level 7] AH02545: fcgi: has determined UDS as /var/run/php-fpm/www.sock
Apache-Error: [file "proxy_util.c"] [line 2615] [level 7] AH00947: connected //usr/local/www/nextcloud/remote.php to httpd-UDS:0
Apache-Error: [file "mod_authz_core.c"] [line 815] [level 7] AH01626: authorization result of Require all granted: granted
Apache-Error: [file "mod_authz_core.c"] [line 815] [level 7] AH01626: authorization result of <RequireAny>: granted
Apache-Handler: proxy:unix:/var/run/php-fpm/www.sock|fcgi://localhost/
Stopwatch: 1607464832497264 1811922 (- - -)
Stopwatch2: 1607464832497264 1811922; combined=6154, p1=850, p2=4233, p3=90, p4=177, p5=502, sr=455, sw=302, l=0, gc=0
Response-Body-Transformed: Dechunked
Producer: ModSecurity for Apache/2.9.3 (http://www.modsecurity.org/); OWASP_CRS/3.3.0.
Server: Apache
Engine-Mode: "ENABLED"

--3974fd1a-Z--

--ac99d975-A--
[09/Dec/2020:01:00:38 +0300] X8-3hi8t@p8dA1rIDm3AHgAAlwY 192.168.0.11 52567 192.168.0.27 443
--ac99d975-B--
PROPFIND /remote.php/dav/files/root/Nextcloud.png HTTP/2.0
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:78.0) Gecko/20100101 Firefox/78.0
Accept: text/plain
Accept-Language: ru-RU,ru;q=0.8,en-US;q=0.5,en;q=0.3
Accept-Encoding: gzip, deflate, br
Requesttoken: Ywq6rZNti+Gbyi8KhOJ5jrJL2zU8K5YYPKc2Si7jh/0=:Vj7+/Pc3uIv/nFU746BLvYAm6FcFZ/JMX8pnImua8oo=
X-Requested-With: XMLHttpRequest
Depth: 0
Content-Type: text/plain;charset=UTF-8
Content-Length: 661
Origin: https://cloud.example.com
Dnt: 1
Cookie: __Host-nc_sameSiteCookielax=true; __Host-nc_sameSiteCookiestrict=true; nc_username=root; nc_token=%2BXZxBY9JfwBMmbmJr8LOA9PyjsCMrI5w; nc_session_id=kiiekiom9u7jb8tdt5vfnhq4nv; oc_sessionPassphrase=jXTF8AHxCsrOuQ9QkW280JAzm8UA7%2BeOJ6vrvzZV20SHXqAq88xgF0XjgB6RkbsD%2B3HdnWJqD8I4femKvxUyt7DldgkpzLQF6k8nnQc9hzK6QAlQUitVFjYiINF5Zq1f; i18next=ru-RU; och0h0q18vvv=kiiekiom9u7jb8tdt5vfnhq4nv
Te: trailers
Host: cloud.example.com

--ac99d975-C--
<?xml version="1.0"?>
			<d:propfind  xmlns:d="DAV:"
				xmlns:oc="http://owncloud.org/ns"
				xmlns:nc="http://nextcloud.org/ns"
				xmlns:ocs="http://open-collaboration-services.org/ns">
				<d:prop>
					<d:getlastmodified />
					<d:getetag />
					<d:getcontenttype />
					<d:resourcetype />
					<oc:fileid />
					<oc:permissions />
					<oc:size />
					<d:getcontentlength />
					<nc:has-preview />
					<nc:mount-type />
					<nc:is-encrypted />
					<ocs:share-permissions />
					<oc:tags />
					<oc:favorite />
					<oc:comments-unread />
					<oc:owner-id />
					<oc:owner-display-name />
					<oc:share-types />
				</d:prop>
			</d:propfind>
--ac99d975-F--
HTTP/1.1 403 Forbidden
Referrer-Policy: no-referrer
X-Content-Type-Options: nosniff
X-Download-Options: noopen
X-Frame-Options: SAMEORIGIN
X-Permitted-Cross-Domain-Policies: none
X-Robots-Tag: none
X-XSS-Protection: 1; mode=block
Content-Length: 199
Connection: close
Content-Type: text/html; charset=iso-8859-1

--ac99d975-E--
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>403 Forbidden</title>
</head><body>
<h1>Forbidden</h1>
<p>You don't have permission to access this resource.</p>
</body></html>

--ac99d975-H--
Message: Access denied with code 403 (phase 2). Pattern match "(?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\\s+(?:\\/|\\w)[^\\s]*(?:\\s+http\\/\\d|[\\r\\n])" at REQUEST_BODY. [file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-921-PROTOCOL-ATTACK.conf"] [line "52"] [id "921110"] [msg "HTTP Request Smuggling Attack"] [data "Matched Data: propfind  xmlns:d=\x22dav:\x22\x0a found within REQUEST_BODY: <?xml version=\x221.0\x22?>\x0a\x09\x09\x09<d:propfind  xmlns:d=\x22dav:\x22\x0a\x09\x09\x09\x09xmlns:oc=\x22http://owncloud.org/ns\x22\x0a\x09\x09\x09\x09xmlns:nc=\x22http://nextcloud.org/ns\x22\x0a\x09\x09\x09\x09xmlns:ocs=\x22http://open-collaboration-services.org/ns\x22>\x0a\x09\x09\x09\x09<d:prop>\x0a\x09\x09\x09\x09\x09<d:getlastmodified />\x0a\x09\x09\x09\x09\x09<d:getetag />\x0a\x09\x09\x09\x09\x09<d:getcontenttype />\x0a..."] [severity "CRITICAL"] [ver "OWASP_CRS/3.3.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag
Apache-Error: [file "ssl_engine_kernel.c"] [line 376] [level 7] AH02034: Subsequent (No.10) HTTPS request received for child 38662 (server cloud.example.com:443)
Apache-Error: [file "mod_authz_core.c"] [line 815] [level 7] AH01626: authorization result of Require all granted: granted
Apache-Error: [file "mod_authz_core.c"] [line 815] [level 7] AH01626: authorization result of <RequireAny>: granted
Apache-Error: [file "apache2_util.c"] [line 271] [level 3] [client 192.168.0.11] ModSecurity: Access denied with code 403 (phase 2). Pattern match "(?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\\\\\\\\s+(?:\\\\\\\\/|\\\\\\\\w)[^\\\\\\\\s]*(?:\\\\\\\\s+http\\\\\\\\/\\\\\\\\d|[\\\\\\\\r\\\\\\\\n])" at REQUEST_BODY. [file "/usr/local/etc/modsecurity/owasp-crs/rules/REQUEST-921-PROTOCOL-ATTACK.conf"] [line "52"] [id "921110"] [msg "HTTP Request Smuggling Attack"] [data "Matched Data: propfind  xmlns:d=\\\\x22dav:\\\\x22\\\\x0a found within REQUEST_BODY: <?xml version=\\\\x221.0\\\\x22?>\\\\x0a\\\\x09\\\\x09\\\\x09<d:propfind  xmlns:d=\\\\x22dav:\\\\x22\\\\x0a\\\\x09\\\\x09\\\\x09\\\\x09xmlns:oc=\\\\x22http://owncloud.org/ns\\\\x22\\\\x0a\\\\x09\\\\x09\\\\x09\\\\x09xmlns:nc=\\\\x22http://nextcloud.org/ns\\\\x22\\\\x0a\\\\x09\\\\x09\\\\x09\\\\x09xmlns:ocs=\\\\x22http://open-collaboration-services.org/ns\\\\x22>\\\\x0a\\\\x09\\\\x09\\\\x09\\\\x09<d:prop>\\\\x0a\\\\x09\\\\x09\\\\x09\\\\x09\\\\x09<d:getlastmodified />\\\\x0a\\\\x09\\\\x09\\\\x09\\\\x09\\\\x09<d:getetag />\\\\x0a\\\\x09\\\\x09\\\\x09\\\\x09\\\\x09<d:getcontenttype />\\\\x0a..."] [severity "CRITICAL"] [ver "OWASP_CRS/3.3.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag [hostname "cloud.example.com"] [uri "/remote.php/dav/files/root/Nextcloud.png"] [unique_id "X8-3hi8t@p8dA1rIDm3AHgAAlwY"]
Action: Intercepted (phase 2)
Stopwatch: 1607464838899234 3303 (- - -)
Stopwatch2: 1607464838899234 3303; combined=2142, p1=834, p2=600, p3=0, p4=0, p5=450, sr=315, sw=258, l=0, gc=0
Response-Body-Transformed: Dechunked
Producer: ModSecurity for Apache/2.9.3 (http://www.modsecurity.org/); OWASP_CRS/3.3.0.
Server: Apache
Engine-Mode: "ENABLED"

--ac99d975-Z--
````

### Audit Logs / Triggered Rule Numbers

The problem occurs when opening files. For example, in the above log, I tried to open the Nextcloud.png file. 
In log I saw, that problem in REQUEST-921-PROTOCOL-ATTACK.conf  - line "52". I commented rule with id:921110 and images and other files are now open.

### Your Environment

* CRS version: 3.3.0
* Paranoia level setting: 1
* ModSecurity version: 2.9.3
* Web Server and version: apache 2.4.46
* Operating System and version: FreeBSD 12.2

### Confirmation

[ x ] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by azurit** [comment]
Can you try it with the newest Nextcloud exclusion rules package?

https://raw.githubusercontent.com/coreruleset/coreruleset/v3.4/dev/rules/REQUEST-903.9003-NEXTCLOUD-EXCLUSION-RULES.conf

**Comment by azurit** [comment]
According to log analyzation, this issue is already fixed in CRS 3.4.0 (currently in development). Thank you for reporting, closing.

**Comment by subzero355** [comment]
Using the latest ruleset, I am getting this and unable to browse the files on Nextcloud.
```
ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/usr/local/modsecurity-crs/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "139"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver "OWASP_CRS/3.4.0-dev"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "1.2.3.4"] [uri "/remote.php/dav/files/xxx/"] [unique_id "1626875708"] [ref ""]
```
